packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx

```diff
@@ -136,19 +176,30 @@ const VisitNotesForm: React.FC<VisitNotesFormProps> = ({ patientUuid, closeWorks
     };
 
     const ac = new AbortController();
-    saveVisitNote(ac, visitNotePayload).then((response) => {
-      if (response.status === 201) {
-        closeWorkspace();
+    saveVisitNote(ac, visitNotePayload)
+      .then((response) => {
+        if (response.status === 201) {
+          closeWorkspace();
+
+          showToast({
+            kind: 'success',
+            description: t('visitNoteSaved', 'Visit note saved successfully'),
+          });
+        }
+      })
+      .catch((err) => {
         showToast({
-          kind: 'success',
-          description: t('visitNoteSaved', 'Visit note saved successfully'),
+          kind: 'error',
+          description: err.message ? err.message : `${err}`,
+          title: t('visitNoteSaveError', 'Error saving visit note'),
         });
-      }
-      response.status !== 201 && createErrorHandler();
-      setIsSubmitting(false);
-    });
+        createErrorHandler();
+      });
 
-    return () => ac.abort();
+    return () => {
+      ac.abort();
```

Comment:
Hm the click handler should not do anything with the returned value (i.e., function in this case). Not sure if this is useful...

Reply:
(we can leave this for now, but again - this is a code smell and we should correct it as soon as possible!)